File: .i3/scripts/wallpaper_rotate.py
#! /usr/bin/python

import os
import time

starttime = time.time()
counter = 0

while True:
    #uses 'feh' to open the next wallpaper in the directory 
    os.system("feh --bg-fill ~/wallpaper/" + str(counter) + ".png")
    
    #after 10 minutes loop back
    time.sleep(600.0 - ((time.time() - starttime)) % 600.0)
    counter += 1
    
    #checks to see if the directory is out of wallpapers, if it is loopback
    if counter > (len(os.listdir('/home/ben/wallpaper')) - 1):
        counter = 0

File: .i3/scripts/wallpaper_rotate.rb
#!/usr/bin/env ruby

counter = 0

loop do
  # feh command to open the next wallpaper
  if !system( "feh --bg-fill ~/wallpaper/" + counter.to_s + ".png" )
    print "Unable to load wallpaper: " + counter.to_s + ".png."
    exit
  end

  # sleep for ~10 minutes
  sleep(600)
  
  # increment wallpaper index
  counter += 1
  
  # loop back to first wallpaper (subtract 1 for 0 index)
  if counter > Dir["**"].length - 1 then counter = 0 end
end
